modules/common/src/test/HtmlTest.scala

```diff
@@ -57,41 +57,41 @@ class HtmlTest extends Specification {
+      addLinks("lichess.org/@/foo/games").body must_==
         """<a href="/@/foo/games">lichess.org/@/foo/games</a>"""
-      addLinks("lichess.org/@/foo") must_== """<a href="/@/foo">@foo</a>"""
-      addLinks("http://lichess.org/") must_== """<a href="/">lichess.org/</a>"""
-      addLinks("http://lichess.org") must_== """<a href="/">lichess.org</a>"""
-      addLinks("@foo") must_== """<a href="/@/foo">@foo</a>"""
+      addLinks("lichess.org/@/foo").body must_== """<a href="/@/foo">@foo</a>"""
+      addLinks("http://lichess.org/").body must_== """<a href="/">lichess.org/</a>"""
+      addLinks("http://lichess.org").body must_== """<a href="/">lichess.org</a>"""
+      addLinks("@foo").body must_== """<a href="/@/foo">@foo</a>"""
     }
 
     "handle trailing punctuation" in {
-      addLinks("lichess.org.") must_== """<a href="/">lichess.org</a>."""
-      addLinks("lichess.org)") must_== """<a href="/">lichess.org</a>)"""
-      addLinks("lichess.org/()") must_== """<a href="/()">lichess.org/()</a>"""
+      addLinks("lichess.org.").body must_== """<a href="/">lichess.org</a>."""
+      addLinks("lichess.org)").body must_== """<a href="/">lichess.org</a>)"""
+      addLinks("lichess.org/()").body must_== """<a href="/()">lichess.org/()</a>"""
 
-      addLinks("lichess.org/())") must_== """<a href="/()">lichess.org/()</a>)"""
-      addLinks("lichess.org/(2)-)?") must_== """<a href="/(2)">lichess.org/(2)</a>-)?"""
+      addLinks("lichess.org/())").body must_== """<a href="/()">lichess.org/()</a>)"""
+      addLinks("lichess.org/(2)-)?").body must_== """<a href="/(2)">lichess.org/(2)</a>-)?"""
 
-      addLinks("lichess.org.-") must_== """<a href="/">lichess.org</a>.-"""
+      addLinks("lichess.org.-").body must_== """<a href="/">lichess.org</a>.-"""
     }
 
     "handle embedded links" in {
-      addLinks(".lichess.org") must_== """.lichess.org"""
-      addLinks("/lichess.org") must_== """/lichess.org"""
-      addLinks(".http://lichess.org") must_== """.<a href="/">lichess.org</a>"""
+      addLinks(".lichess.org").body must_== """.lichess.org"""
+      addLinks("/lichess.org").body must_== """/lichess.org"""
+      addLinks(".http://lichess.org").body must_== """.<a href="/">lichess.org</a>"""
 
-      addLinks("/http://lichess.org") must_== """/<a href="/">lichess.org</a>"""
+      addLinks("/http://lichess.org").body must_== """/<a href="/">lichess.org</a>"""
     }
 
     "handle ambig path separator" in {
-      addLinks("lichess.org#f") must_== """<a href="/#f">lichess.org/#f</a>"""
-      addLinks("lichess.org?f") must_== """<a href="/?f">lichess.org/?f</a>"""
+      addLinks("lichess.org#f").body must_== """<a href="/#f">lichess.org/#f</a>"""
+      addLinks("lichess.org?f").body must_== """<a href="/?f">lichess.org/?f</a>"""
     }
 
     "pass through plain text (fast case)" in {
       val noUrl = "blah blah foobar"
-      addLinks(noUrl) must be(noUrl)
+      addLinks(noUrl).body must_== noUrl
```

Comment:
this checks that the returned object is the same instance, not string equality.

Reply:
yes, i was aware of that and it would be no longer guaranteed after this patch